src/main/java/ui/components/pickers/RepositoryPickerState.java

```diff
@@ -0,0 +1,128 @@
+package ui.components.pickers;
+
+import org.controlsfx.control.spreadsheet.Picker;
+
+import java.util.*;
+import java.util.stream.IntStream;
+
+/**
+ * A class used to store the list of repositories, used by RepositoryPicker.
+ */
+public class RepositoryPickerState {
+
+    private final List<PickerRepository> repositories = new ArrayList<>();
+    private final List<PickerRepository> matchingRepositories = new ArrayList<>();
+
+    public RepositoryPickerState(Set<String> storedRepositories) {
+        storedRepositories.stream()
+                .forEach(repo -> repositories.add(new PickerRepository(repo)));
+        Collections.sort(repositories);
+    }
+
+    public void updateUserQuery(String query) {
+        clearRepositorySelection();
+        updateMatchingRepositories(query);
+    }
```

Comment:
Public methods should have comments, also because `updateUserQuery` may not be clear enough to describe what it does.